### ISSUE TITLE
Async Context Managers

### ISSUE THREAD

**Issue by rlkandela** [pylint-dev/pylint#10604]
### Question

When I declare an async context manager in an interface I do it the following way
```python
@abstractmethod
def connect(self) -> contextlib.AbstractAsyncContextManager[AsyncConnection]:
   pass
```

Then I implement it using the asynccontextmanager decorator
```python
@contextlib.asynccontextmanager
async def connect(self) -> AsyncIterator[AsyncConnection]: ...
```

But pylint complains about it being an invalid override due to the sync-async difference. Is there a better/correct way of doing this or I should just tell pylint to ignore the warning?

### Documentation for future user

May be a snippet next to the [W0236:invalid-overridden-method](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/invalid-overridden-method.html) as it's the warning it gives when doing so.

**Comment by mbyrnepr2** [comment]
Similar to #8939 and #5761 and #9745
